test/e2e/common/utils.py

```diff
@@ -36,65 +40,60 @@
 TRANSFORMER_CONTAINER = "transformer-container"
 STORAGE_URI_ENV = "STORAGE_URI"
 
+rest_client_v1 = None
+rest_client_v2 = None
+grpc_client = None
+
+configure_logging()
+
 
-def grpc_stub(service_name, namespace):
+def grpc_stub(host):
     cluster_ip = get_cluster_ip()
-    method_config = json.dumps(
-        {
-            "methodConfig": [
-                {
-                    "name": [{"service": "org.pytorch.serve.grpc.inference"}],
-                    "retryPolicy": {
-                        "maxAttempts": 5,
-                        "initialBackoff": "0.1s",
-                        "maxBackoff": "10s",
-                        "backoffMultiplier": 2,
-                        "retryableStatusCodes": ["UNAVAILABLE"],
-                    },
-                }
-            ]
-        }
-    )
+    if ":" not in cluster_ip:
+        cluster_ip = cluster_ip + ":80"
+    logger.info("Cluster IP: %s", cluster_ip)
+    logger.info("gRPC target host: %s", host)
     os.environ["GRPC_VERBOSITY"] = "debug"
-    channel = grpc.insecure_channel(
-        cluster_ip,
-        options=(
-            (
-                "grpc.ssl_target_name_override",
-                service_name + "." + namespace + ".example.com",
-            ),
-            ("grpc.service_config", method_config),
-        ),
-    )
-    return inference_pb2_grpc.InferenceAPIsServiceStub(channel)
-
+    if grpc_client is None:
+        return InferenceGRPCClient(
+            cluster_ip,
+            verbose=True,
+            channel_args=[
+                ("grpc.ssl_target_name_override", host),
+            ],
+        )
+    return grpc_client
 
-def predict(
-    service_name,
-    input_json,
-    protocol_version="v1",
-    version=constants.KSERVE_V1BETA1_VERSION,
-    model_name=None,
-):
-    with open(input_json) as json_file:
-        data = json.load(json_file)
 
-        return predict_str(
-            service_name=service_name,
-            input_json=json.dumps(data),
-            protocol_version=protocol_version,
-            version=version,
-            model_name=model_name,
-        )
+def get_rest_client(protocol):
+    global rest_client_v1
+    global rest_client_v2
```

Comment:
Let's avoid using global variables, you can create the client in a test fixture and pass to the test function